torchao/experimental/_linear_8bit_act_xbit_weight_subclass_quantizer.py

```diff
@@ -0,0 +1,408 @@
+        return cls(packed_weight, scale, zero_point, layout_type)
+
+    def _apply_fn_to_data(self, fn):
+        self.packed_weight = fn(self.packed_weight)
+        if self.scale is not None:
+            self.scale = fn(self.scale)
+
+        if self.zero_point is not None:
+            self.zero_point = fn(self.zero_point)
+        return self
+
+    @classmethod
+    def __torch_dispatch__(cls, func, types, args, kwargs):
+        kwargs = {} if kwargs is None else kwargs
+
+        if func is torch.ops.aten.detach.default:
+            return return_and_correct_aliasing(
+                func, args, kwargs, args[0]._apply_fn_to_data(torch.detach)
+            )
+        if func is torch.ops.aten.clone.default:
+            return return_and_correct_aliasing(
+                func, args, kwargs, args[0]._apply_fn_to_data(torch.clone)
+            )
+
+        raise NotImplementedError(
+            f"IntxWeightAQTLayout dispatch: attempting to run {func}, this is not supported"
+        )
+
+    def __tensor_flatten__(self):
+        if _target_equal(self.get_layout_type().target, "native"):
+            return ["packed_weight"], [self.get_layout_type()]
+
+        # fallback
+        assert _target_equal(self.get_layout_type().target, "fallback")
+        if self.zero_point is None:
+            return ["packed_weight", "scale"], [self.get_layout_type()]
+        return ["packed_weight", "scale", "zero"], [self.get_layout_type()]
+
+    @classmethod
+    def __tensor_unflatten__(
+        cls, tensor_data_dict, tensor_attributes, outer_size, outer_stride
+    ):
+        packed_weight, scale, zero_point = (
+            tensor_data_dict["packed_weight"],
+            tensor_data_dict.get("scale", None),
+            tensor_data_dict.get("zero_point", None),
+        )
+        (layout_type,) = tensor_attributes
+        return cls(packed_weight, scale, zero_point, layout_type)
+
+
+def _linear_int8_dynamic_activation_intx_weight_check(
+    input_tensor, weight_tensor, bias
+):
+    layout_type = weight_tensor.layout_tensor.layout_type
+    return isinstance(layout_type, IntxWeightLayoutType) and bias is None
+
+
+def _linear_int8_dynamic_activation_intx_weight_fallback_impl(
+    input_tensor, weight_tensor, bias
+):
+    assert weight_tensor.layout_tensor.layout_type.target == "fallback"
+    assert bias is None
+
+    def _impl_2d(input_tensor, weight_tensor):
+        assert input_tensor.dim() == 2
+        assert weight_tensor.dim() == 2
+
+        weight_qvals = weight_tensor.layout_tensor.packed_weight.to(torch.int32)
+        weight_scales = weight_tensor.layout_tensor.scale
+        weight_zeros = weight_tensor.layout_tensor.zero_point
+        group_size = weight_tensor.layout_tensor.layout_type.group_size
+        has_weight_zeros = weight_zeros is not None
+        m, k = input_tensor.shape
+        n, k_ = weight_tensor.shape
+        assert k_ == k
+
+        weights_dequantized = dequantize_per_channel_group(
```

Comment:
we tend to use quantize_affine/dequantize_affine I think, also this should probably be:

`weights_dequantized = weight_tensor.dequantize()`?